lib/engine-jit/src/engine.rs

```diff
@@ -157,7 +157,7 @@ pub struct JITEngineInner {
     features: Features,
     /// The code memory is responsible of publishing the compiled
     /// functions to memory.
-    code_memory: CodeMemory,
+    code_memory: Vec<CodeMemory>,
```

Comment:
I'm trying to understand the needs/implications of this. I guess now each artifact will have its own `CodeMemory`?
Is that assumption accurate?

Reply:
Yes, and each `CodeMemory` has one `Mmap`. To get small code model to work, everything a compiled module needs is within a single 32-bit region.